### PR TITLE
fix(wireguard): prevent error when playbook run is limited to specific wg peers

### PR DESCRIPTION
**Pull request by blueorca363**

### CODE REVIEW

File: roles/wireguard/templates/iface_config.conf.j2 [hunk 1]
```diff
@@ -17,28 +17,42 @@ Table = {{ item.table }}
...
+AllowedIPs = {{
+    (
       (hostvars[peer].wg_ifaces
       | selectattr('iface_name', 'equalto', item.iface_name)
-      | map(attribute='peering_port')
-      | first)
-    ] | join(':')
+      | map(attribute='ip_addresses_cidr')
+      | first
+      + (item.allowed_ips | default([])))
```
Comment (lukaslihotzki-f) [inline roles/wireguard/templates/iface_config.conf.j2:30]: Please indent either the `|` or `+` lines. Currently, the precedence is not obvious.

File: roles/wireguard/templates/iface_config.conf.j2 [hunk 2]
```diff
@@ -17,28 +17,42 @@ Table = {{ item.table }}
...
+      | list
+      | join(', ')
+    ) | default(
+      (wireguard_config_directory[peer][item.iface_name].ip_addresses_cidr
+      + (item.allowed_ips | default([])))
+      | list
+      | join(', ')
```
Comment (lukaslihotzki-f) [inline roles/wireguard/templates/iface_config.conf.j2:37]: Can the join be applied once, to the final list, after the fallback to wireguard_config_directory? So, `(hostvars | … | default(wiregr)) | join(', ')`.

**Review by lukaslihotzki-f — CHANGES_REQUESTED** [review]

File: roles/wireguard/templates/iface_config.conf.j2 (hunk 1 above)
Comment (blueorca363) [reply roles/wireguard/templates/iface_config.conf.j2:30]: Fixed locally.

File: roles/wireguard/templates/iface_config.conf.j2 (hunk 2 above)
Comment (blueorca363) [reply roles/wireguard/templates/iface_config.conf.j2:37]: Trying, this *should* work in theory.

Comment (lukaslihotzki-f) [reply roles/wireguard/templates/iface_config.conf.j2:37]: This change seems to be done in `Endpoint` only (where `join(':')`), but not in `AllowedIPs` (where `join(', ')`).

Comment (blueorca363) [reply roles/wireguard/templates/iface_config.conf.j2]: Oooops, sorry. You were right. Fixed.

**Review by lukaslihotzki-f — DISMISSED** [review]

**Review by lrsksr — CHANGES_REQUESTED** [review]
Could you add documentation about that new feature in the README.md of the role? The second paragraph needs to be updated to reflect your changes.

File: roles/wireguard/templates/iface_config.conf.j2 [hunk 3]
```diff
@@ -17,28 +17,40 @@ Table = {{ item.table }}
...
+    (
+      (hostvars[peer].wg_ifaces
+        | selectattr('iface_name', 'equalto', item.iface_name)
+        | map(attribute='ip_addresses_cidr')
+        | first
+      + (item.allowed_ips | default([])))
+      | list
+    ) | default(
+      (wireguard_config_directory[peer][item.iface_name].ip_addresses_cidr
+      + (item.allowed_ips | default([])))
```
Comment (lukaslihotzki-f) [inline roles/wireguard/templates/iface_config.conf.j2:34]: `item.allowed_ips` will always be empty when this is reached. If it was non-empty, this default is not reached. Therefore, it has no effect to add this here. Probably, you want to factor out the addition of `item.allowed_ips` like the addition of `join` was factored out.

Comment (blueorca363) [reply roles/wireguard/templates/iface_config.conf.j2]: I optimized the Jinja2 code accordingly and also made all indents consistent with the rest as well.

**Review by lukaslihotzki-f — APPROVED** [review]

**Pull request merged**